server/services/roService.ts

```diff
@@ -22,8 +24,25 @@ export class RoService {
     }
   }
 
-  private filterMostRecentActiveBooking(offenderSentences: OffenderSentence[]): OffenderSentence[] {
-    return offenderSentences.filter((o) => o.mostRecentActiveBooking === true)
+  getLatestSentences(offenderSentences: OffenderSentence[]): OffenderSentence[] {
+    const groupedOffenderSentencess = groupBy(offenderSentences, ({ offenderNo }) => offenderNo)
+    let latestOffenderSentences = []
+
+    Array.from(groupedOffenderSentencess.values()).flatMap((sentences) => {
+      const hasNoDatesToCompare = sentences.find((b) => absentDatesToCompare(b))
+      if (hasNoDatesToCompare) {
+        sentences.forEach((sentence) => latestOffenderSentences.push(sentence))
```

Comment:
**todo:** `flatmap` is used to flatten a list of responses which means we can avoid mutation.
The way it's used here currently is to solely do the iteration (it could be replaced by a `forEach`)  

to use `flatMap` more effectively you'd assign the result of the flatMap:

`const latestOffenderSentences  = Array.from(groupedOffenderSentencess.values()).flatMap((sentences)...`

and then in each branch would return an array you want to flatten into the single returned array
so on this line:  `return sentences`
and in the other branch: `return [sortedSentences.pop()]`

Reply:
Thank you. Refactored to use `flatMap` more effectively as suggested. Stupid question please - why do we return the last sorted sentence (`sortedSentences.pop()`) in an array?